File: output/mistralai_mistral_small_2603/zero/python/Arquivos.py

import sys

def main():
    data = sys.stdin.read().split()
    n = int(data[0])
    B = int(data[1])
    sizes = list(map(int, data[2:2+n]))
    sizes.sort()
    left = 0
    right = n - 1
    folders = 0
    while left <= right:
        if sizes[left] + sizes[right] <= B:
            left += 1
        right -= 1
        folders += 1
    print(folders)

if __name__ == "__main__":
    main()

File: output/mistralai_mistral_small_2603/zero/cpp/Arquivos.cpp

#include <bits/stdc++.h>
using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr);

    int N;
    long long B;
    cin >> N >> B;

    vector<long long> files(N);
    for (int i = 0; i < N; ++i) {
        cin >> files[i];
    }

    sort(files.begin(), files.end());

    int left = 0;
    int right = N - 1;
    int folders = 0;

    while (left <= right) {
        if (files[left] + files[right] <= B) {
            left++;
            right--;
        } else {
            right--;
        }
        folders++;
    }

    cout << folders << '\n';
    return 0;
}
